google-cloud-firestore/CONTRIBUTING.md

```diff
@@ -1,187 +1,390 @@
+   above):
+
+   ``` sh
+   $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/keyfile.json
+   ```
+
+   If you are already using the `GOOGLE_APPLICATION_CREDENTIALS` environment variable, and wish to test this library
+   with a different key file, you may set the `FIRESTORE_CREDENTIALS` environment variable instead:
+
+   ``` sh
+   $ export FIRESTORE_CREDENTIALS=/path/to/keyfile.json
+   ```
+
+1. Set the `GOOGLE_CLOUD_PROJECT` environment variable to your Google Cloud project ID:
+
+   ``` sh
+   $ export GOOGLE_CLOUD_PROJECT=my-project-id
+   ```
+
+   If you are already using the `GOOGLE_CLOUD_PROJECT` environment variable, and wish to test this library with a
+   different project, you may set the `FIRESTORE_PROJECT` environment variable instead:
+
+   ``` sh
+   $ export FIRESTORE_PROJECT=my-project-id
+   ```
+
+
+## Run CI
+
+You are now ready to run local CI checks for the library, which you should do **before** you make any changes. Doing so
+ensures that everything is OK with your local environment and the latest dependency versions. You don't want any
+surprises later.
+
+To run the code style checks, documentation tests, and unit tests together, use the `ci` task:
 
 ``` sh
-$ cd google-cloud-firestore/
 $ bundle exec rake ci
 ```
 
-To run the command above, plus all acceptance tests, use `rake ci:acceptance` or
-its handy alias, `rake ci:a`.
+To run the command above, plus all acceptance tests, use `rake ci:acceptance` or its handy alias, `rake ci:a`. Keep in
+mind that the acceptance tests typically take longer than the other CI checks.
 
-### Firestore Unit Tests
+The Rake tasks aggregated in the commands above can be run individually to streamline your workflow when developing or
+debugging.
 
+| CI check                                      | Command           |
+|-----------------------------------------------|------------------ |
+| [Static code analysis](#Static-code-analysis) | `rake rubocop`    |
+| [Documentation tests](#Documentation-tests)   | `rake doctest`    |
+| [Unit tests](#Unit-tests)                     | `rake test`       |
+| [Acceptance tests](#Acceptance-tests)         | `rake acceptance` |
 
-The project uses the [minitest](https://github.com/seattlerb/minitest) library,
-including [specs](https://github.com/seattlerb/minitest#specs),
-[mocks](https://github.com/seattlerb/minitest#mocks) and
-[minitest-autotest](https://github.com/seattlerb/minitest-autotest).
+The subsections below describe the individual CI checks.
 
-To run the Firestore unit tests:
+### Static code analysis
 
-``` sh
-$ cd google-cloud-firestore/
-$ bundle exec rake test
+The project uses [Rubocop](https://github.com/rubocop/rubocop) configured with the shared
+[googleapis/ruby-style](https://github.com/googleapis/ruby-style) rules to ensure that your code adheres to
+Google's Ruby style. The style is largely based on [The Ruby Style
+Guide](https://github.com/bbatsov/ruby-style-guide) with a few exceptions based on Seattle.rb style:
```

Comment:
Link to https://seattlerb.org/

Reply:
I certainly considered this link, and tried hard to find any link to an authoritative description of  Seattle.rb style. Do you know of one?

Reply:
Good question. I'm not sure, but lemme ask in the SeattleRB slack and see if anyone else knows.

Reply:
Fixed.